app/static/src/app/components/WodinPanels.vue

```diff
@@ -0,0 +1,186 @@
+<template>
+    <div :class="modeClass">
+        <div class="wodin-left">
+            <span class="wodin-collapse-controls">
+                <span v-if="canCollapseLeft" @click="collapseLeft()" class="collapse-control" id="collapse-left">
+                    <vue-feather type="chevron-left"></vue-feather>
+                </span>
+                <span v-if="canCollapseRight" @click="collapseRight()" class="collapse-control" id="collapse-right">
+                    <vue-feather type="chevron-right"></vue-feather>
+                </span>
+            </span>
+            <div class="view-left" @click="collapseRight()">
+                View Options
+            </div>
+            <div class="wodin-content" id="wodin-content-left">
+                <slot name="left"></slot>
+            </div>
+        </div>
+        <div class="wodin-right">
+            <div class="view-right" @click="collapseLeft()">
+                View Charts
+            </div>
+            <div class="wodin-content" id="wodin-content-right">
+                <slot name="right"></slot>
+            </div>
+        </div>
+    </div>
+    <div style="clear:both;"></div>
+</template>
+
+<script lang="ts">
+import { computed, defineComponent, ref } from "vue";
+import VueFeather from "vue-feather";
+
+export enum PanelsMode {
+        Left, Right, Both
+    }
+
+export default defineComponent({
+    name: "WodinPanels",
+    components: {
+        VueFeather
+    },
+    setup() {
+        const mode = ref<PanelsMode>(PanelsMode.Both);
+
+        const modeClass = computed(() => {
+            switch (mode.value) {
+            case PanelsMode.Left:
+                return "wodin-mode-left";
+            case PanelsMode.Both:
+                return "wodin-mode-both";
+            case PanelsMode.Right:
+                return "wodin-mode-right";
+            default:
+                throw new Error("Unknown mode");
```

Comment:
Might be able to save a few lines by referencing an object rather than using a switch statement.

Reply:
Ah yeah, you tend to do that rather than using enums don't you? I think I prefer the enum -I find it clearer as to intent, though switch statements are a bit verbose as you say...